src/www/app/outline_server_repository/index.ts

```diff
@@ -53,11 +53,20 @@ function isDynamicAccessKey(accessKey: string): boolean {
 // NOTE: For extracting a name that the user has explicitly set, only.
 // (Currenly done by setting the hash on the URI)
 function serverNameFromAccessKey(accessKey: string): string {
-  if (isDynamicAccessKey(accessKey)) {
-    return new URL(accessKey.replace(/^ssconf:\/\//, 'https://')).hostname;
+  const {hostname, hash} = new URL(accessKey.replace(/^ss(?:conf)?:\/\//, 'https://'));
+
+  let result;
+  if (hash && hash !== '#') {
+    // locates the first key-value pair in the search params-compliant
+    // string that has no value (i.e. thing=a&My%20Server%20Name&thing=b)
+    const {groups} = hash.slice(1).match(/(?:^|&)(?<name>[^=]+)(?:&|$)/);
+
+    result = groups?.name;
   }
 
-  return SHADOWSOCKS_URI.parse(accessKey).tag.data;
+  result ??= hostname;
```

Comment:
Note that this will make the behavior consistent for static and dynamic keys, in that the default server name will be the ip/domain embedded within the key.

Reply:
This is an abstraction layer violation.
`serverNameFromAccessKey` shouldn't return a name if it's not in the access key. That's a format/standard functionality.

Put the "default" name logic in the caller. That's an app-specific functionality.

Reply:
Still pending